csm_api_client/service/hsm.py

```diff
@@ -243,8 +262,8 @@ def get_node_components(self, ancestor=None):
         err_prefix = 'Failed to get Node components'
         try:
             if ancestor:
-                ancestor = XName(ancestor)
-                if not ancestor.is_valid:
+                ancestor_xname = XName(ancestor)
+                if not ancestor_xname.is_valid:
                     raise APIError(f'Could not get descendants of {ancestor}: invalid xname')
 
                 components = self.get('State', 'Components', 'Query', str(ancestor), params={'type': 'Node'}).json()
```

Comment:
Not sure it matters, but should `str(ancestor)` be `str(ancestor_xname)` now?

Reply:
Both should be equivalent, though for the sake of maintaining the implementation I'll change it to `ancestor_xname`.